curvefs/src/client/metaserver_client.cpp

```diff
@@ -17,65 +17,301 @@
+CURVEFS_ERROR MetaServerClientImpl::CreateInode(const InodeParam &param,
+                                                Inode *out) {
+    auto task = RPCTaskDefine {
+        CreateInodeResponse response;
+        basecli_->CreateInode(param, &response, cntl, channel);
+
+        if (cntl->Failed()) {
+            LOG(WARNING) << "CreateInode Failed, errorcode = "
+                         << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        MetaStatusCode stcode = response.statuscode();
+        MetaServerStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "CreateInode:  param = "
+            << ", errcode = " << retcode
+            << ", errmsg = " << MetaStatusCode_Name(stcode);
+
+        if (response.has_inode()) {
+            *out = response.inode();
+        }
+        return static_cast<int>(retcode);
+    };
+    return excutor_.DoRPCTask(task);
 }
 
-CURVEFS_ERROR MetaServerClientImpl::DeleteInode(
-    uint32_t fsId, uint64_t inodeid) {
-    return CURVEFS_ERROR::OK;
+CURVEFS_ERROR MetaServerClientImpl::DeleteInode(uint32_t fsId,
+                                                uint64_t inodeid) {
+    auto task = RPCTaskDefine {
+        DeleteInodeResponse response;
+        basecli_->DeleteInode(fsId, inodeid, &response, cntl, channel);
+
+        if (cntl->Failed()) {
+            LOG(WARNING) << "DeleteInode Failed, errorcode = "
+                         << cntl->ErrorCode()
+                         << ", error content:" << cntl->ErrorText()
+                         << ", log id = " << cntl->log_id();
+            return -cntl->ErrorCode();
+        }
+
+        CURVEFS_ERROR retcode = CURVEFS_ERROR::FAILED;
+        MetaStatusCode stcode = response.statuscode();
+        MetaServerStatusCode2CurveFSErr(stcode, &retcode);
+        LOG_IF(WARNING, retcode != CURVEFS_ERROR::OK)
+            << "DeleteInode:  fsid = " << fsId << ", inodeid = " << inodeid
+            << ", errcode = " << retcode
+            << ", errmsg = " << MetaStatusCode_Name(stcode);
+
+        return static_cast<int>(retcode);
+    };
+    return excutor_.DoRPCTask(task);
 }
 
-CURVEFS_ERROR MetaServerClientImpl::AllocExtents(uint32_t fsId,
-    const std::list<ExtentAllocInfo> &toAllocExtents,
+void MetaServerClientImpl::MetaServerStatusCode2CurveFSErr(
+    const MetaStatusCode &statcode, CURVEFS_ERROR *errcode) {
+    switch (statcode) {
+    case MetaStatusCode::OK:
+        *errcode = CURVEFS_ERROR::OK;
+        break;
+    case MetaStatusCode::PARAM_ERROR:
+        *errcode = CURVEFS_ERROR::INVALIDPARAM;
+        break;
+    default:
+        *errcode = CURVEFS_ERROR::UNKNOWN;
+        break;
+    }
+}
+
+
+CURVEFS_ERROR MetaServerClientImpl::AllocExtents(
```

Comment:
Delete this two function?

Reply:
Deal with your commit, the current framework still has references